### DISCUSSION TITLE
Override link elements from a parent when using DynamicLinksFunction in a child route

### DISCUSSION THREAD

**Discussion by scottwilkin** [sergiodxa/remix-utils#177] [Q&A] [answered]
I'm implementing `DynamicLinksFunction` in an app that needs to use `loader` data in both the root, and child routes. If both the root and the child set a link with the same `rel` value e.g. `image_src` you end up with two elements with different `href` values. If `DynamicLinksFunction` included `matches` in the args, similar to the V2 MetaFunction: https://remix.run/docs/en/1.14.0/route/meta#matches then you could create a util function to merge the `LinkDescriptor[]` values while de-duping.

**Comment by sergiodxa** [comment, marked as answer]
DynamicLinksFunction is being deprecated in favor of the v2 of MetaFunction, I recommend checking how that works instead of using DynamicLinksFunction.
